pointerevents/pointerevent_touch-action-keyboard.html

```diff
@@ -78,66 +78,66 @@ <h4 id="desc">Test Description: Press DOWN ARROW key. Wait for description updat
+            var x_scroll_baseline, y_scroll_baseline, x_scroll, y_scroll;
 
-            var xScrollIsReceived = false;
-            var yScrollIsReceived = false;
-            var xScr0, yScr0, xScr1, yScr1;
+            var got_x_scroll = false;
+            var got_y_scroll = false;
 
-            function run() {
-                var target0 = document.getElementById("target0");
+            async function send_key_to_target(key_code) {
+                return new test_driver.Actions()
+                    .addKeyboard("keyboard1", "keyboard")
+                    .keyDown(key_code, {origin: target0})
+                    .keyUp(key_code, {origin: target0})
+                    .send();
+            }
 
+            async function run() {
                 var test_touchaction = async_test("touch-action attribute test");
-                var actions_promise;
 
-                xScr0 = target0.scrollLeft;
-                yScr0 = target0.scrollTop;
+                const target0 = document.getElementById("target0");
+                const x_scroll_baseline = target0.scrollLeft;
+                const y_scroll_baseline = target0.scrollTop;
 
-                target0.focus();
+                // Inject keyboard scroll inputs.
+                const arrow_down = "\uE015";
+                const arrow_right = "\uE014";
 
                 on_event(target0, 'scroll', function(event) {
-                    xScr1 = target0.scrollLeft;
-                    yScr1 = target0.scrollTop;
-
-                    if(xScr1 != xScr0) {
-                        xScrollIsReceived = true;
-                    }
-
-                    if(yScr1 != yScr0) {
+                    x_scroll = target0.scrollLeft;
+                    y_scroll = target0.scrollTop;
+                    if (y_scroll > y_scroll_baseline && !got_y_scroll) {
                         test_touchaction.step(function () {
-                            yScrollIsReceived = true;
-                            assert_true(true, "y-scroll received.");
+                            assert_true(true, "content was scrolled down.");
                         });
-                        updateDescriptionNextStepKeyboard();
+                        got_y_scroll = true;
                     }
-
-                    if(xScrollIsReceived && yScrollIsReceived) {
-                        // Make sure the test finishes after all the input actions are completed.
-                        actions_promise.then( () => {
-                            test_touchaction.done();
+                    if (x_scroll > x_scroll_baseline && !got_x_scroll) {
+                        test_touchaction.step(function () {
+                            assert_true(true, "content was scrolled right.");
                         });
+                        got_x_scroll = true;
+                    }
+                    if (got_x_scroll && got_y_scroll) {
                         updateDescriptionComplete();
```

Comment:
I would omit this line from the past.  TestDriver based tests are not manually testable.